provider/throttle_utils.py

```diff
@@ -141,3 +152,600 @@ def change_throttle_group(self, image, group_id):
+        """
+
+        :param test: Context of test.
+        :param params: params of running ENV.
+        :param vm: VM object.
+        :param session: Session object connect to guest.
+        :param group: Throttle group name.
+        :param images: list of relevant images names.
+        """
+
+        self._test = test
+        self._vm = vm
+        self._session = session
+        self._monitor = vm.monitor
+        self._fio = None
+        self._params = params
+        self.group = group
+        # shared fio option without --filename
+        self._fio_option = ""
+        self.images = images.copy() if images else []
+        self._throttle = {
+            "images": {image: copy.deepcopy(ThrottleTester.raw_image_data) for
+                       image in images},
+            "expected": copy.deepcopy(ThrottleTester.raw_expected)}
+        self._margin = 0.3
+
+    @staticmethod
+    def generate_output_by_json(output):
+        """
+        Convert fio command output to dict object.
+
+        :param output: fio command output with option --output-format=json.
+        :return: dict of fio command output.
+        """
+
+        with tempfile.TemporaryFile(mode="w+") as tmp:
+            tmp.write(output)
+            tmp.seek(0)
+            line = tmp.readline()
+            begin_flag = False
+            block_index = 1
+            block = {}
+            data = ""
+            while line:
+                if line == "{\n":
+                    if not begin_flag:
+                        begin_flag = True
+                    else:
+                        # error
+                        break
+                if begin_flag:
+                    data += line
+                else:
+                    # skip
+                    pass
+                if line == "}\n":
+                    if begin_flag:
+                        begin_flag = False
+                    else:
+                        # error
+                        break
+                    block[block_index] = json.loads(data)
+                    data = ""
+                    block_index += 1
+
+                line = tmp.readline()
+
+            if begin_flag:
+                print("error1")
+            return block
+
+    def set_fio(self, fio):
+        """
+        Set fio instance.
+
+        :param fio: fio instance.
+        """
+
+        self._fio = fio
+
+    def run_fio(self, *args):
+        """
+        Start to fio command in guest.
+
+        :param args: image data,data struct refer to raw_image_data.
+        :return: fio command output.
+        """
+
+        if not self._fio:
+            self._test.error("Please set fio first")
+        image_info = args[0]
+        fio_option = image_info["fio_option"]
+        session = self._vm.wait_for_login()
+        cmd = ' '.join((self._fio.cfg.fio_path, fio_option))
+        cmd += " && " + cmd
+        logging.info("run_fio:" + cmd)
+        out = session.cmd(cmd, 1800)
```

Comment:
The cmd is customised, it integrated two fio command. image_info["fio_option"] is not the full fio command,so it need to run cmd instead of image_info["fio_option"]